### PR TITLE
Make set do more things

### PR DESCRIPTION
**Pull request by meffij**

### CODE REVIEW

File: api/python/t4/snapshots.py [hunk 1]
```diff
@@ -216,6 +216,21 @@ def as_dict(self):
...
+                'type': 'SHA256',
+                'value': hash_file(file_to_hash)
+            }
+
+        size = os.path.getsize(path)
+        physical_keys = [{
+            'type': PhysicalKeyType.LOCAL.name,
+            'path': os.path.abspath(path)
+        }]
+        return PackageEntry(physical_keys, size, hash_obj, {})
```
Comment (stevededalus) [inline api/python/t4/snapshots.py]: Add a todo to ingest metadata here when https://github.com/quiltdata/t4/pull/28 is ready

File: api/python/t4/snapshots.py [hunk 2]
```diff
@@ -400,9 +404,22 @@ def update(self, logical_key, entry):
         Returns:
             A new package
         """
+        if 'meta' in kwargs and entry is None:
+            return self._update_meta(logical_key, kwargs['meta'])
+
+        pkg = self._clone()
+        if isinstance(entry, str):
```
Comment (stevededalus) [inline api/python/t4/snapshots.py]: Update the docstring arg type to reflect that it can be a string now.

File: api/python/t4/snapshots.py [hunk 3]
```diff
@@ -400,9 +404,22 @@ def update(self, logical_key, entry):
         Returns:
             A new package
         """
+        if 'meta' in kwargs and entry is None:
```
Comment (stevededalus) [inline api/python/t4/snapshots.py]: Just make meta a named optional keyword arg in the signature

File: api/python/t4/snapshots.py [hunk 4]
```diff
@@ -400,9 +404,22 @@ def update(self, logical_key, entry):
...
+            return self._update_meta(logical_key, kwargs['meta'])
+
```
Comment (stevededalus) [inline api/python/t4/snapshots.py]: raise an exception if entry and meta are both missing?

File: api/python/t4/snapshots.py (hunk 2 above)
Comment (meffij) [reply api/python/t4/snapshots.py]: done

**Pull request merged**